addons/web/static/src/core/notifications/notification.xml

```diff
@@ -26,6 +24,9 @@
                 </button>
                 </div>
             </div>
+            <button type="button" class="o_notification_close close ml-2 position-absolute" aria-label="Close" t-on-click="props.close">
+                <span class="d-inline-block" aria-hidden="true">×</span>
+            </button>
```

Comment:
I see your point on a design point of view.
But I feel like, at this point, it becomes unproductive to use the Bootstrap Toast component as we diverge (DOM isn't the same, positioning is float-like but with `absolute`...) too much from it.

Note that the Bootstrap documentation told us:
> To encourage extensible and predictable toasts, we recommend a header and body. Toast headers use display: flex, allowing easy alignment of content thanks to our margin and flexbox utilities.

Reply:
Bootstrap is meant to be customised, it's what xlu did.
I don't see the problem of using the framework we include in the bundle.
I see a design improvement and ultimately less code.

Reply:
For me there is a difference between customising the component using SCSS variables provided by the framework plus a few utility classes and modifying the DOM structure and/or element positioning of the component because we want something-like-that-but-not-really. Also it has a maintainability impact when upgrading the framework.

Reply:
It's what we already did for dropdowns, modals, buttons, etc.. 😉
If the component has a corresponding bootstrap one, then we use it. 

In this case too, we're not using the `.toast` class to define the design of a modal (for example).
An odoo notification is ultimately a bootstrap toast indeed.

This may lead to some overrides, but it's what happens when you rely on an hyper-declarative framework. 
Maybe in the feature bootstrap will provide us more variables... or we'll create our own framework.
Until that day "the right class to the right place", I guess.  😉

> modifying the DOM structure

I don't see any problem as long as it doesn't leads to the invalidation of the component's purpose or semantic (eg. a navbar-tab without tabs, `.btn` class applied to a `<nav>` tag ). 
 
 > and/or element positioning 
 
 I think that toast position is meant to be adapted.

Reply:
As we are not totally agree with this point, could we let it unchanged and either create another task to improve it or wait for complains... What do you think?

Reply:
@xlu-odoo , I guess you can just move the "close icon" into `.toast-header`, remove its `position-absolute` and  remove the `t-if="props.title"` from `.toast-header`.
This should give us a 100% compliant toast component.

@pparidans , is it ok for you?

Reply:
Yes(-ish).

Well, notifications will now require a title-text to be designed correctly, the final result will be a bit off otherwise.
@xlu-odoo solution is elegant and has a limited impact IMO, but if it raises implementation problem I'm ok with the change.

I expect "complains" about  it (probably what @adr-odoo  was referring to?), but we can always review this behaviour later 👍

Reply:
@pparidans , there is actually a JS test to ensure that  a notification can be rendered without title :/ (https://github.com/odoo/odoo/blob/15.0/addons/web/static/tests/webclient/actions/client_action_tests.js#L501).

It's a common scenario and something we should take in account.
I suggest to restore back to the initial implementation using position-absolute on the button.

This is not good :/
![image](https://user-images.githubusercontent.com/8558351/155155563-a57207e6-33d3-4e54-a48f-36404c313444.png)

Reply:
> It's a common scenario and something we should take in account. I suggest to restore back to the initial implementation using position-absolute on the button.

@stefanorigano Then what would you think about not using the `.toast` component at all (as we diverge from it's base implementation anyway)?

Something like:
```diff
diff --git a/addons/web/static/src/core/notifications/notification.scss b/addons/web/static/src/core/notifications/notification.scss
index a004cafe9e5..1219fd5ab95 100644
--- a/addons/web/static/src/core/notifications/notification.scss
+++ b/addons/web/static/src/core/notifications/notification.scss
@@ -3,11 +3,16 @@
 // The notification just add overlays.
 
 .o_notification_manager {
-  @include o-position-absolute($o-navbar-height * 1.15, map-get($spacers, 2), $left: calc(100vw - #{$toast-max-width}));
+  @include o-position-absolute($o-navbar-height * 1.15, map-get($spacers, 2), $left: calc(100vw - #{$o-notification-max-width}));
   z-index: $zindex-notification;
 
   .o_notification {
     border-left-width: $o-notification-border-left-width !important;
+    box-shadow: $o-notification-shadow;
+  }
+
+  .o_notification_close {
+    @include o-position-absolute(map-get($spacers, 2), map-get($spacers, 2));
   }
 }
 
diff --git a/addons/web/static/src/core/notifications/notification.variables.scss b/addons/web/static/src/core/notifications/notification.variables.scss
index 1b73a3a72a2..bf890ce8e0d 100644
--- a/addons/web/static/src/core/notifications/notification.variables.scss
+++ b/addons/web/static/src/core/notifications/notification.variables.scss
@@ -1,3 +1,4 @@
+$o-notification-max-width: 320px;
 $o-notification-border-left-width: 0.75rem !default;
 $o-notification-shadow: 0 12px 14px -10px rgba(0, 0, 0, .25);
 $zindex-notification: 1055 !default;
diff --git a/addons/web/static/src/core/notifications/notification.xml b/addons/web/static/src/core/notifications/notification.xml
index e517b368fd6..3cf37456d83 100644
--- a/addons/web/static/src/core/notifications/notification.xml
+++ b/addons/web/static/src/core/notifications/notification.xml
@@ -2,18 +2,14 @@
 <templates xml:space="preserve">
 
     <t t-name="web.NotificationWowl" owl="1">
-        <div t-attf-class="o_notification toast show border-{{props.type}} overflow-hidden mb-2"
+        <div t-attf-class="o_notification border border-{{props.type}} bg-white mb-2"
         role="alert" aria-live="assertive" aria-atomic="true">
-            <div t-attf-class="toast-header align-items-start pb-0">
-                <strong t-attf-class="o_notification_title text-{{props.type}} mr-auto" t-esc="props.title"/>
-                <button type="button" class="o_notification_close close ml-2" aria-label="Close" t-on-click="props.close">
-                    <span class="d-inline-block" aria-hidden="true">×</span>
-                </button>
-            </div>
-            <div t-attf-class="o_notification_body toast-body px-3 py-2">
-                <div t-if="props.message" class="mr-auto o_notification_content">
-                    <t t-out="props.message"/>
-                </div>
+            <strong t-if="props.title" t-attf-class="o_notification_title d-block text-{{props.type}} py-2 pl-3 pr-4" t-out="props.title"/>
+            <button type="button" class="o_notification_close close" aria-label="Close" t-on-click="props.close">
+                <span class="d-block" aria-hidden="true">×</span>
+            </button>
+            <div t-attf-class="o_notification_body pl-3 pr-4 py-2">
+                <div t-if="props.message" class="mr-auto o_notification_content" t-out="props.message"/>
                 <div t-if="props.buttons.length" class="mt-2 o_notification_buttons">
                     <button t-foreach="props.buttons" t-as="button" type="button" t-key="button_index"
                     t-attf-class="btn {{button.primary ? 'btn-primary' : 'btn-link'}}" t-on-click="button.onClick">
```

Reply:
![unknown](https://user-images.githubusercontent.com/80678921/155505475-85bf704d-6809-4f00-b0cf-87f770ba6cb2.png)
PR Updated with toast removed

Reply:
Looks good to me, your feedback @stefanorigano ?

Reply:
Yes, sorry for the delay.

Design is good except for the broken transitions.
Since we "unlinked" from boostrap `.toast` component  I suggest to exclude it from bootstrap's compilation.... I see no point on include/maintain 2 different notification design.

Reply:
> Design is good except for the broken transitions.
Yes also seen it...

> Since we "unlinked" from boostrap `.toast` component I suggest to exclude it from bootstrap's compilation.... I see no point on include/maintain 2 different notification design.

Agree with that. Just need to be sure we don't use it anywhere else in Odoo before trimming it of the bundle.

Reply:
> Agree with that. Just need to be sure we don't use it anywhere else in Odoo before trimming it of the bundle.
Yes, there would be some occurrence to handle, eg
https://github.com/odoo/enterprise/blob/master/planning/views/planning_templates.xml#L150

Reply:
I won't mind since we don't especially use them but it feels like the most meaningful argument to keep a proper bootstrap structure was left out of this conversation :wink: If a custom website theme now customizes toasts, that won't impact website toasts :shrug: Again, I don't mind since we don't use them but I would have kept using bootstrap classes no matter what :wink:

Reply:
> Again, I don't mind since we don't use them but I would have kept using bootstrap classes no matter what

As you said we don't use them a lot, and I'm not sure how much we want these components to be customised by a theme... for sure we don't want themes to impact the editorUI, I guess.

But if we extend this component to be used by the "visitor UI" too (eg. "_Added to chart_", "_Email sent_" etc...)[1], then these last notifications should be customised by the theme, I agree 👍 

Maybe the new-coming "editor into <iframe>" architecture will be handy by this point of view (one design for the editor and another one for the visitor) .. but not using `.toast` classes here invalidates the logic, I agree.

---

[1] maybe is it already the case?